### PR TITLE
Add changes from rector rule SimplifyIfReturnBoolRector

### PR DESCRIPTION
**Pull request by Harfusha**
https://github.com/cakephp/cakephp/issues/17756

### CODE REVIEW

**Comment by dereuromark** [comment]
The value here seems small compared to the merge conflicts arising from 5.x to 5.next probably

**Comment by ADmad** [comment]
Since we are nearing 5.1 stable, we shouldn't have too many merge issues in the future. 

The merge conflict in the PR needs to be fixed though :)

File: src/Datasource/EntityTrait.php [hunk 1]
```diff
@@ -466,18 +466,13 @@ public function has(array|string $field): bool
...
-            return true;
-        }
 
-        return false;
+        return $value === null ||
+        (
+            is_array($value) &&
+            empty($value) ||
+            $value === ''
+        );
```
Comment (othercorey) [inline src/Datasource/EntityTrait.php:475]: Neither of these is great. Think this is a place where multiple return paths is easier.

File: src/Error/Debug/HtmlFormatter.php [hunk 2]
```diff
@@ -53,11 +53,7 @@ public function __construct()
      */
     public static function environmentMatches(): bool
     {
-        if (PHP_SAPI === 'cli' || PHP_SAPI === 'phpdbg') {
-            return false;
-        }
-
-        return true;
+        return !(PHP_SAPI === 'cli' || PHP_SAPI === 'phpdbg');
```
Comment (othercorey) [inline src/Error/Debug/HtmlFormatter.php:56]: Surprised this one isn't turned into != && != like the other PR.

File: src/Validation/Validator.php [hunk 3]
```diff
@@ -3130,15 +3130,9 @@ protected function isEmpty(mixed $data, int $flags): bool
...
-            && $data instanceof UploadedFileInterface
-            && $data->getError() === UPLOAD_ERR_NO_FILE
-        ) {
-            return true;
-        }
-
-        return false;
+        return ($flags & self::EMPTY_FILE)
+        && $data instanceof UploadedFileInterface
+        && $data->getError() === UPLOAD_ERR_NO_FILE;
```
Comment (othercorey) [inline src/Validation/Validator.php:3135]: We should indent the && lines below return.

File: src/Error/Debug/HtmlFormatter.php (hunk 2 above)
Comment (Harfusha) [reply src/Error/Debug/HtmlFormatter.php:56]: maybe it will, after the rules will run at a same time

**Comment by Harfusha** [comment]
I dont see much value in this either, will close this PR

**Pull request closed without merging**